plugins/cue_points/annotation/lib/api/KalturaAnnotation.php

```diff
@@ -197,37 +197,47 @@ public function updateEndTimeAndDuration($cuePoint)
 	}
 
 	/*
-     * @param string $cuePointId
-     * @throw KalturaAPIException - when parent annotation doesn't belong to the same entry
-     */
-    public function validateParentId($cuePointId = null)
-    {
+	 * @param string $cuePointId
+	 * @throw KalturaAPIException - when parent annotation doesn't belong to the same entry
+	 */
+	public function validateParentId($cuePointId = null)
+	{
 		//Backward compatibility patch
 		if ($this->isNull('parentId'))
 			$this->parentId = 0;
 
-        if ($this->parentId)
-        {
-            $dbParentCuePoint = CuePointPeer::retrieveByPK($this->parentId);
-            if (!$dbParentCuePoint)
-                throw new KalturaAPIException(KalturaCuePointErrors::PARENT_ANNOTATION_NOT_FOUND, $this->parentId);
+		if ($this->parentId)
+		{
+			$dbParentCuePoint = CuePointPeer::retrieveByPK($this->parentId);
+			if (!$dbParentCuePoint)
+				throw new KalturaAPIException(KalturaCuePointErrors::PARENT_ANNOTATION_NOT_FOUND, $this->parentId);
 
-            if($cuePointId !== null){// update
-                $dbCuePoint = CuePointPeer::retrieveByPK($cuePointId);
-                if(!$dbCuePoint)
-                    throw new KalturaAPIException(KalturaCuePointErrors::INVALID_OBJECT_ID, $cuePointId);
+			if($cuePointId !== null)
+			{// update
+				$dbCuePoint = CuePointPeer::retrieveByPK($cuePointId);
+				if(!$dbCuePoint)
+					throw new KalturaAPIException(KalturaCuePointErrors::INVALID_OBJECT_ID, $cuePointId);
 
-                if($dbCuePoint->isDescendant($this->parentId))
-                    throw new KalturaAPIException(KalturaCuePointErrors::PARENT_ANNOTATION_IS_DESCENDANT, $this->parentId, $dbCuePoint->getId());
+				if($dbCuePoint->isDescendant($this->parentId))
+					throw new KalturaAPIException(KalturaCuePointErrors::PARENT_ANNOTATION_IS_DESCENDANT, $this->parentId, $dbCuePoint->getId());
+
+				if ($dbParentCuePoint->getEntryId() != $dbCuePoint->getEntryId())
+					throw new KalturaAPIException(KalturaCuePointErrors::PARENT_ANNOTATION_DO_NOT_BELONG_TO_THE_SAME_ENTRY);
+			}
+			else
+			{
+				if ($dbParentCuePoint->getEntryId() != $this->entryId)
+					throw new KalturaAPIException(KalturaCuePointErrors::PARENT_ANNOTATION_DO_NOT_BELONG_TO_THE_SAME_ENTRY);
+			}
+		}
+	}
 
-                if ($dbParentCuePoint->getEntryId() != $dbCuePoint->getEntryId())
-                    throw new KalturaAPIException(KalturaCuePointErrors::PARENT_ANNOTATION_DO_NOT_BELONG_TO_THE_SAME_ENTRY);
-            }
-            else
-            {
-                if ($dbParentCuePoint->getEntryId() != $this->entryId)
-                    throw new KalturaAPIException(KalturaCuePointErrors::PARENT_ANNOTATION_DO_NOT_BELONG_TO_THE_SAME_ENTRY);
-            }
-        }
-    }
+	protected function validateEntryEntitlement(entry $dbEntry, $tags)
+	{
+		if (!kEntitlementUtils::isEntryEntitled($dbEntry) && str_contains($tags, 'hotspots'))
```

Comment:
If I have a cue-point with the following tags:
- `coolhotspots`
- `tag2`
- `tag3`

which translates into `coolhotspots,tag2,tag3` in the DB (and in the API entry object) - this restriction will apply there, which is unexpected.
I think that you should do something like

```php
$tags = explode(',', $tags);
if (in_array('hotspots', $tags) && !kEntitlementUtils::isEntryEntitled($dbEntry)) {
    // throw error
}
```

Also notice the order of the conditions - `isEntryEntitled` will not execute if `hotspots` isn't one of the tags, which might be performance improvement.

Reply:
Opened a new PR #12969